f/frizzle/comapeo/comapeo_observations.py

```diff
@@ -92,20 +108,50 @@ def fetch_comapeo_projects(
             if project["project_id"] not in comapeo_project_blocklist
         ]
 
-    if not comapeo_projects:
-        raise ValueError("No valid projects found.")
-
     return comapeo_projects
 
 
+@retry(
+    stop_max_attempt_number=3,
+    wait_exponential_multiplier=500,
+)
+def download_attachment(url, headers, save_path):
+    try:
+        response = requests.get(url, headers=headers)
+
+        content_type = response.headers.get("Content-Type", "")
+        extension = mimetypes.guess_extension(content_type) or ""
+
+        file_name = os.path.basename(url) + extension
+        os.makedirs(os.path.dirname(save_path), exist_ok=True)
+        with open(save_path, "wb") as file:
+            file.write(response.content)
+        logger.info("Download completed.")
+        return file_name
+
+    except Exception as e:
+        logger.error(f"Exception during download: {e}")
+        return None
+
+
+def camel_to_snake(name):
+    return re.sub("([a-z0-9])([A-Z])", r"\1_\2", name).lower()
+
+
 def process_comapeo_data(
-    comapeo_server_base_url, comapeo_access_token, comapeo_projects, attachment_root
+    comapeo_server_base_url,
+    comapeo_access_token,
+    comapeo_projects,
+    db_table_prefix,
+    attachment_root,
+    attachment_failed,
 ):
     comapeo_data = {}
     for index, project in enumerate(comapeo_projects):
         project_id = project["project_id"]
         project_name = project["project_name"]
         sanitized_project_name = re.sub(r"\W+", "_", project_name).lower()
+        prefixed_project_name = f"{db_table_prefix}_{sanitized_project_name}"
```

Comment:
Maybe can wait for later, but would we ever allow the `db_table_prefix` to be empty? In which case we should not prepend the `_`.